docs/src/guidelines.md

```diff
@@ -0,0 +1,97 @@
+# [Guidelines for creating models](@id Guidelines)
+
+These are guidelines for the creation of models using NLPModels to help keeping the models uniform, and for future reference in the creation of solvers.
+
+## Bare minimum
+
+Your model should derive from `AbstractNLPModel` or some other abstract class derived from it.
+It is mandatory that it has a `meta :: NLPModelMeta` field, storing all the relevant problem information.
+The minimum information that should be set is
+
+- `nvar`: Number of variables
+- `x0`: Starting point (defaults to `zeros`)
+- `lvar`, `uvar`: Bounds on the variables (default to `(-∞,∞)`)
+- `ncon`: Number of constraints (defaults to `0`)
+- `lcon`, `ucon`: Bounds on the constraints (default to `(-∞,∞)`)
+- `nnzh`: The length of the vectors used to store a triangle of the Hessian matrix in triplet format (defaults to `nvar * (nvar + 1) / 2`
+- `nnzj`: The length of the vectors used to store the Jacobian matrix in triplet format (default to `nvar * ncon`)
+
+The model also need to provide `Counters` information, the easiest way being defining `counters :: Counters`.
+For instance:
+```julia
+mutable struct MyModel <: AbstractNLPModel
+  meta :: NLPModelMeta
+  counters :: Counters
+end
```

Comment:
the part about counters and this example should be moved above what needs to go into meta.

Reply:
Between lines 8 and 9?